docs/guides/gitlab.md

```diff
@@ -0,0 +1,280 @@
+  :::
+
+- The initial setup documented here uses un-encrypted HTTP traffic between Pomerium and GitLab. The last section covers upgrading the GitLab configuration with a TLS certificate.
+
+## Install GitLab
+
+::: tip
+While we do our best to keep our documentation up to date, changes to third-party systems are outside our control. Refer to [GitLab Docker Images] from GitLab's docs as needed, or [let us know](https://github.com/pomerium/pomerium/issues/new?assignees=&labels=&template=bug_report.md) if we need to re-visit this section.
+:::
+
+### Prepare The Environment
+
+1.  Create volumes for persistent data. This guide uses the base directory `/srv/gitlab`; adjust this path for your local environment:
+
+    ```bash
+    mkdir -p /srv/gitlab #Adjust the path based on your common Docker volume location.
+    ```
+
+1.  Create three sub-directories: `config`, `data`, and `logs`:
+
+    ```bash
+    mkdir -p /srv/gitlab/{config,data,logs}
+    ```
+
+### Install and Configure GitLab
+
+1.  Edit your `docker-compose.yml` file to define a new service for GitLab:
+
+    ```yaml
+    services:
+
+    ...
+
+    gitlab:
+      container_name: gitlab
+      image: gitlab/gitlab-ee:latest
+      environment:
+        GITLAB_OMNIBUS_CONFIG: |
+          external_url 'https://gitlab.pomerium.localhost.io'
+          registry_external_url 'https://gitlab.pomerium.localhost.io'
+          letsencrypt['enable'] = false
+          nginx['listen_port'] = 80
+          nginx['listen_https'] = false
+          registry_nginx['listen_port'] = 80
+          registry_nginx['listen_https'] = false
+          pages_nginx['listen_port'] = 80
+          pages_nginx['listen_https'] = false
+          mattermost_nginx['listen_port'] = 80
+          mattermost_nginx['listen_https'] = false
+      volumes:
+        - '/srv/gitlab/config:/etc/gitlab'
+        - '/srv/gitlab/logs:/var/log/gitlab'
+        - '/srv/gitlab/data:/var/opt/gitlab'
+      expose:
+        - 80
+        - 443
+        - 22
+      restart: always
+      shm_size: '256m'
+    ```
+
+    - Adjust `external_url` and `registry_external_url` to match the external path, which we will define in Pomerium later in the process.
+    - Adjust the paths under `volumes` to match the directories created in [the previous section](#prepare-the-environment).
+
+1.  Bring up the new Docker Compose configuration:
+
+    ```bash
+    docker-compose up -d
+    ```
+
+    The container may take several minutes to initialize. Once complete, the status provided to `docker ps` will change from `health:starting` to `healthy`.
+    
+    You can also monitor the progress with `docker logs -f gitlab`. Note that even after the process is complete, the log file will continue to output log messages.
+
+## Configure a Pomerium Route
+
+Edit `config.yaml` and add a route for GitLab. Note that this example assumes access to the mkcert CA certificate at `/mkcert/rootCA.pem`:
```

Comment:
> Note that this example assumes access to the mkcert CA certificate at `/mkcert/rootCA.pem`:

This appears outdated or moved to the TLS config section?

Reply:
```suggestion
Edit `config.yaml` and add a route for GitLab:
```